File: go/pkg/util/cards.go
package util

import (
	"bufio"
	"os"
	"strconv"
	"strings"
)

type Cards struct {
	filename string
	cardname []string
	addnum   int
}

func NewCards(filename string, cardname []string, addnum int) *Cards {
	return &Cards{
		filename: filename,
		cardname: cardname,
		addnum:   addnum,
	}
}

func (c *Cards) Get() []string {
	file, err := os.OpenFile(c.filename, os.O_RDONLY|os.O_CREATE, 0644)
	if err != nil {
		return []string{}
	}
	file.Close()

	content, err := os.ReadFile(c.filename)
	if err != nil {
		return []string{}
	}

	rawstr := strings.TrimSpace(string(content))
	if rawstr == "" {
		return []string{}
	}

	rawvec := strings.Split(rawstr, ",")
	seen := make(map[int]bool)
	var realCardVec []string

	for _, item := range rawvec {
		item = strings.TrimSpace(item)
		if !strings.HasPrefix(item, "/x/") {
			continue
		}

		realstr := item[3:]
		num, err := strconv.Atoi(realstr)
		if err != nil {
			continue
		}

		adjustedNum := num - c.addnum
		if seen[adjustedNum] {
			continue
		}
		seen[adjustedNum] = true

		if adjustedNum >= 0 && adjustedNum < len(c.cardname) {
			realCardVec = append(realCardVec, c.cardname[adjustedNum])
		}
	}

	return realCardVec
}

func (c *Cards) Write(index int) int {
	if index < 0 || index >= len(c.cardname) {
		return -1
	}

	file, err := os.OpenFile(c.filename, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
	if err != nil {
		return -1
	}
	defer file.Close()

	writer := bufio.NewWriter(file)
	_, err = writer.WriteString(",/x/" + strconv.Itoa(c.addnum+index))
	if err != nil {
		return -1
	}
	writer.Flush()

	return 0
}

func (c *Cards) GetByIndex(index int) string {
	if index < 0 || index >= len(c.cardname) {
		return ""
	}
	return c.cardname[index]
}

File: lib/tool/cards.dart
import 'dart:io';

class Cards {
  final String filename;
  final List<String> cardname;
  final int addnum;

  Cards({
    required this.filename,
    required this.cardname,
    required this.addnum,
  });

  List<String> get() {
    final file = File(filename);

    try {
      // 确保文件存在
      if (!file.existsSync()) {
        file.createSync(recursive: true);
        return [];
      }

      final content = file.readAsStringSync().trim();
      if (content.isEmpty) {
        return [];
      }

      final rawVec = content.split(',');
      final seen = <int>{};
      final realCardVec = <String>[];

      for (var item in rawVec) {
        item = item.trim();
        if (!item.startsWith('/x/')) {
          continue;
        }

        final realstr = item.substring(3);
        final num = int.tryParse(realstr);
        if (num == null) {
          continue;
        }

        final adjustedNum = num - addnum;
        if (seen.contains(adjustedNum)) {
          continue;
        }
        seen.add(adjustedNum);

        if (adjustedNum >= 0 && adjustedNum < cardname.length) {
          realCardVec.add(cardname[adjustedNum]);
        }
      }

      return realCardVec;
    } catch (e) {
      print('Error reading cards file: $e');
      return [];
    }
  }

  int write(int index) {
    if (index < 0 || index >= cardname.length) {
      return -1;
    }

    try {
      final file = File(filename);
      if (!file.existsSync()) {
        file.createSync(recursive: true);
      }

      final sink = file.openSync(mode: FileMode.append);
      sink.writeStringSync(',/x/${addnum + index}');
      sink.flushSync();
      sink.closeSync();

      return 0;
    } catch (e) {
      print('Error writing to cards file: $e');
      return -1;
    }
  }

  String getByIndex(int index) {
    if (index < 0 || index >= cardname.length) {
      return '';
    }
    return cardname[index];
  }
}
